pkg/subctl/cmd/validate_fwconfig.go

```diff
@@ -0,0 +1,184 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/submariner-io/submariner-operator/apis/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/pkg/internal/cli"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/resource"
+)
+
+var validateIntraClusterFWConfig = &cobra.Command{
+	Use:   "fw-intra-cluster",
+	Short: "Validate the Firewall Configuration within the cluster.",
+	Long:  "This command checks whether firewall configuration allows traffic via vx-submariner interface.",
+	Run:   validateFWConfig,
+}
+
+var validationTimeout uint
+
+const (
+	TCPSniffCommand = "tcpdump -ln -c 3 -i vx-submariner tcp and port 8080 and 'tcp[tcpflags] == tcp-syn'"
+)
+
+func addValidateFWConfigFlags(cmd *cobra.Command) {
+	cmd.Flags().UintVar(&validationTimeout, "validation-timeout", 90,
+		"timeout in seconds while validating the connection attempt")
+}
+
+func init() {
+	addValidateFWConfigFlags(validateIntraClusterFWConfig)
+	validateCmd.AddCommand(validateIntraClusterFWConfig)
+}
+
+func validateFWConfig(cmd *cobra.Command, args []string) {
+	configs, err := getMultipleRestConfigs(kubeConfig, kubeContext)
+	exitOnError("Error getting REST config for cluster", err)
+
+	for _, item := range configs {
+		status.Start(fmt.Sprintf("Retrieving Submariner resource from %q", item.clusterName))
+		submariner := getSubmarinerResource(item.config)
+		if submariner == nil {
+			status.QueueWarningMessage(submMissingMessage)
+			status.End(cli.Success)
+			continue
+		}
+
+		status.End(cli.Success)
+		retVal := validateFWConfigWithinCluster(item, submariner)
+		if retVal.HasFailureMessages() {
+			status.End(cli.Failure)
+			continue
+		} else {
+			status.End(cli.Success)
+		}
```

Comment:
I since added `ResultFromMessages`.  Also you're mixing the global Status and the one returned from `validateFWConfigWithinCluster`. But `validateFWConfigWithinCluster` uses the global one anyway so it doesn't matter. So no need to return it.

```suggestion
                 status.End(status.ResultFromMessages())
```